### DISCUSSION TITLE
define theme with css variables

### DISCUSSION THREAD

**Discussion by herzaso** [storybookjs/storybook#25092] [Help]
### Summary

I have a theme file in which I defined colors using hex and rgb but I now moved to defining the colors using CSS variables (in the form of `rgb(var(--text-primary))` where `--text-primary: 51 51 51`) and I get the following error:
`Uncaught Error: Couldn't parse the color string. Please provide the color as a string in hex, rgb, rgba, hsl or hsla notation.`.

Am I doing it wrong? Is it not supported?

**Comment by ShaunEvening** [comment]
Hey @herzaso 👋 

Can you provide more context on where this error is thrown? If you can create a quick reproduction through https://storybook.new I can see what's happening

**Reply by herzaso** [reply]
Sure, here is the reproduction - https://stackblitz.com/edit/github-zrpecs?file=.storybook%2FYourTheme.ts
The only change there is the addition of the theme file from [here](https://storybook.js.org/docs/configure/theming).

If you switch the `colorSecondary` from the hex value to the CSS variable you'll see the following error in your console:
```
Uncaught Error: Couldn't parse the color string. Please provide the color as a string in hex, rgb, rgba, hsl or hsla notation.


    at parseToRgb (chunk-IQUMAILS.js:188:8129)
    at transparentize (chunk-IQUMAILS.js:188:13420)
    at chunk-IQUMAILS.js:188:12566
    at chunk-IQUMAILS.js:190:4281
    at chunk-IQUMAILS.js:190:4413
    at handleInterpolation (chunk-AMRE34KO.js:294:3191)
    at serializeStyles (chunk-AMRE34KO.js:299:1657)
    at chunk-AMRE34KO.js:304:1379
    at chunk-AMRE34KO.js:299:2937
    at renderWithHooks (chunk-AMRE34KO.js:92:907)
    ```

**Reply by ShaunEvening** [reply]
Awesome! Thank you so much for taking the time to put that together for me.

Ah! Now I see what you mean.

We're currently using `polished` to parse and validate colors from the given object. Unfortunately polished has a separate function to support css variables or a default which means it would mean a change in the logic of our theming package.

The good news though is that we're actively looking to move the manager theme customization over to purely use css variables which would sidestep this whole problem. However, this won't happen until Storybook 8, which we've just started working on.

Let me take a quick peek at our open issues that we could link this thread to. If not I'll convert this into an issue for us to add to our backlog.

**Reply by panzacoder** [reply]
@ShaunEvening Did this wind up landing in Storybook 8? I'm using css-variables to define my theme in a SB8 project now and am hitting the same error. 

Oddly, I am able to specify a variable for the `appContentBg` variable, but not the `textColor` or several others.

**Reply by clukhei** [reply]
@ShaunEvening I am in need of the support for css variable for theme customisation too in storybook 8. Any updates on the status ?

**Reply by sipkoepp** [reply]
Same here ...

**Reply by realwitt** [reply]
Ran into this as well. 
Tailwindcss v4 using native css variables makes the reality of pulling all colors from one central spot agonizingly close too. I couldn't use the OKLCH definitions of my color variables either, I had to convert them to hex. ☹️

**Reply by jannikbuscha** [reply]
The restriction still exists in Storybook 9 (tested for Angular):
```ts
export default create({
  ...
  textColor: 'var(--color-text)',
});
```

Will result in:
Couldn't parse the color string. Please provide the color as a string in hex, rgb, rgba, hsl or hsla notation.

But I have found a workaround:
```ts
export default create({
  ...
  appBg: getComputedStyle(document.documentElement).getPropertyValue('--color-green'),
  appContentBg: getComputedStyle(document.documentElement).getPropertyValue('--color-blue'),
  textColor: getComputedStyle(document.documentElement).getPropertyValue('--color-red'),
});
```

docs.scss:
```css
:root {
  --color-green: green;
  --color-blue: #0000ff;
  --color-red: rgb(255, 0, 0);
}
```

which is loaded globally:
```json
"storybook": {
      "executor": "@storybook/angular:start-storybook",
      "options": {
        ...
        "styles": ["libs/shared/ui/.storybook/docs.scss"]
      },
```

Will result in:
<img width="1053" height="303" alt="image" src="https://github.com/user-attachments/assets/af5cca23-998c-43b5-91e2-623a7fbb83f1" />
